app/server/middleware/search.js

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const searchArticles = (request, response, next) => {
+  // only searches a single source, rather than an
+  // array of sources
+  // also doesn't allow sortBy filtering yet, as some
+  // sources don't support all types of filtering
+  const { source } = request.query;
+
+  const search = `https://newsapi.org/v1/articles?apiKey=${process.env.NEWS_KEY}&source=${source}`;
+
+  axios.get(search)
+    .then((data) => {
+      const articles = data.data.articles.map((article) => {
+        const newArticle = article;
+        newArticle.source = source;
+        return article;
+      });
+      request.articles = articles;
+      next();
+    })
+    .catch((err) => {
+      console.log(err);
+      response.status(500);
```

Comment:
is a 500 code the right type of error for getting no data back from an API? I have no idea myself, just wanting to double check

Reply:
500 error codes indicate a server-side issue, rather than a client error (4xx) or redirection (3xx).